src/test/ui/impl-trait/unactionable_diagnostic.rs

```diff
@@ -0,0 +1,23 @@
+trait Trait {}
+
+struct Foo;
+
+impl Trait for Foo {}
+
+fn foo<'t, P>(
+    post: P,
+    x: &'t Foo,
+) -> &'t impl Trait {
```

Comment:
```suggestion
fn foo<'x, P>(
    post: P,
    x: &'x Foo,
) -> &'x impl Trait {
```
If the lifetime here has a different name the error incorrectly suggest adding `+ 't` bound. This should be fixed by removing the `needs_infer` hack as I suggested earlier.